paasta_tools/marathon_tools.py

```diff
@@ -1370,16 +1370,15 @@ def get_expected_instance_count_for_namespace(
     total_expected = 0
     if not cluster:
         cluster = load_system_paasta_config().get_cluster()
-    for name, instance in get_service_instance_list(
-        service,
-        cluster=cluster,
-        instance_type='marathon',
+
+    pscl = PaastaServiceConfigLoader(
+        service=service,
         soa_dir=soa_dir,
-    ):
-        srv_config = load_marathon_service_config(name, instance, cluster, soa_dir=soa_dir)
-        instance_ns = srv_config.get_nerve_namespace()
-        if namespace == instance_ns:
-            total_expected += srv_config.get_instances()
+        load_deployments=False,
+    )
+    for job_config in pscl.instance_configs(cluster=cluster, instance_type_class=MarathonServiceConfig):
+        if f"{service}.{namespace}" in job_config.get_registrations():
```

Comment:
This is great, and I think fixes a kind of long-standing bug of not counting canaries in the expected count!